### PR TITLE
Create driver_timer

### PR DESCRIPTION
**Pull request by OwenV1**
This script is set to notify the driver of the time of the match.  it has an autonomous section at the beginning, and it blinks red for the last 15-ish seconds.

### CODE REVIEW

File: driver_timer [hunk 1]
```diff
@@ -0,0 +1,471 @@
+#include <FastLED.h>
+
+#define NUM_LEDS 60
+#define LED_PIN 2
+#define DATA_PIN 7
+
+void showStrip() {
```
Comment (yelisetti) [inline driver_timer]: Would add some comments as to what this function does?

File: driver_timer [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+FastLED.show();
+}
+
+void setAll(byte red, byte green, byte blue) {
```
Comment (yelisetti) [inline driver_timer]: Comment asking for what the method does?

File: driver_timer [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+  }
+  showStrip();
+}
+
+CRGB leds[NUM_LEDS];
+void setup() {
+FastLED.addLeds<NEOPIXEL, LED_PIN>(leds, NUM_LEDS);
+}
+
+void setPixels(int Pixels, byte red, byte green, byte blue){
```
Comment (yelisetti) [inline driver_timer]: What does this method do? I would add a comment

File: driver_timer [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+void setAll(byte red, byte green, byte blue) {
+  for(int i = 0; i < NUM_LEDS; i++ ){
+  setPixels(i, red, green, blue);
+    
+  }
+  showStrip();
+}
+
+CRGB leds[NUM_LEDS];
+void setup() {
```
Comment (yelisetti) [inline driver_timer]: Give some spacing from the previous line

File: driver_timer [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+  
+  //1st second
+
+  setAll(0x00, 0x00, 0xff);
+  delay(1000);
+  
+//for(int f = NUM_LEDS; f > 0; f-4 ){
+  //setPixels(f, 0x00, 0x00, 0x00);
+//}
+leds[60] = CRGB::Black;
```
Comment (yelisetti) [inline driver_timer]: I would prefer if we are able to write this into a for loop??

File: driver_timer [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+FastLED.show();
+leds[23] = CRGB::Black;
+FastLED.show();
+leds[22] = CRGB::Black;
+FastLED.show();
+leds[21] = CRGB::Black;
+FastLED.show();
+delay(1000);
+ 
+  //change led color to orange
```
Comment (yelisetti) [inline driver_timer]: We are not changing the color to orange? Comment should be deleted

File: driver_timer [hunk 7]
```diff
@@ -0,0 +1,471 @@
...
+  //showStrip();
+  //delay(500);
+  //setAll(0xff, 0x00, 0x00);
+  //showStrip();
+  //delay(500);
+  //setAll(0x00, 0x00, 0x00);
+  //showStrip();
+  //delay(500);
+
+  setAll(0x00, 0xff, 0x00);
```
Comment (yelisetti) [inline driver_timer]: add a comment that the color will be se to green

File: driver_timer [hunk 8]
```diff
@@ -0,0 +1,471 @@
...
+//  leds[7] = CRGB::Black;
+//  FastLED.show();
+//  delay(2250);
+//  leds[6] = CRGB::Black;
+//  FastLED.show();
+//  delay(2250);
+
+  //blink red
+    
+  for(int o = 60; o >= 6; o--){
```
Comment (yelisetti) [inline driver_timer]: Comment what the line does?

File: driver_timer [hunk 9]
```diff
@@ -0,0 +1,471 @@
...
+  //showStrip();
+  //delay(500);
+  //setAll(0x00, 0x00, 0x00);
+  //showStrip();
+  //delay(500);
+
+  setAll(0x00, 0xff, 0x00);
+  delay(2250);
+  
+  //for (int q = 0; q < 1; q++){
```
Comment (yelisetti) [inline driver_timer]: I would take out the code below

File: driver_timer [hunk 10]
```diff
@@ -0,0 +1,471 @@
...
+    //for(int p = 0; p < NUM_LEDS; p = p + 1){
+      //setPixels(p+q, 0, 0, 0);
+      //}
+//  }
+//  leds[60] = CRGB::Black;
```
Comment (yelisetti) [inline driver_timer]: delete the code below, we already have a for loop that does this already?

File: driver_timer [hunk 11]
```diff
@@ -0,0 +1,471 @@
...
+    }
+  
+for(int r = 5; r >= 0; r--){
+    leds[r] = CRGB::Red;
+    FastLED.show();
+    delay(2250);
+  }
+
+  
+//  leds[5] = CRGB::Black;
```
Comment (yelisetti) [inline driver_timer]: Remove code... seems redundant

File: driver_timer [hunk 12]
```diff
@@ -0,0 +1,471 @@
...
+
+  for(int w = 5; w >= 0; w--){
+//    leds[r] = CRGB::Red;
+//    FastLED.show();
+//    delay(2250);
+     setPixels(w, 0xff, 0x00, 0x00);
+     
+    }
+  
+for(int r = 5; r >= 0; r--){
```
Comment (yelisetti) [inline driver_timer]: I would comment as to what this line does.

**Review by yelisetti — COMMENTED** [review]
Reviewed Code.

File: driver_timer [hunk 13]
```diff
@@ -0,0 +1,471 @@
...
+    leds[r] = CRGB::Red;
```
Comment (yelisetti) [inline driver_timer]: This should be set to black. We already set it to Red, before counting down!

**Review by yelisetti — COMMENTED** [review]
small comment added

File: driver_timer (hunk 5 above)
Comment (OwenV1) [reply driver_timer]: I did make one, and it worked, but I can't get it right now because I'm using a different computer

File: driver_timer (hunk 1 above)
Comment (OwenV1) [reply driver_timer]: What do you mean by this?

Comment (yelisetti) [reply driver_timer]: Adding a comment in the code itself about what the function does , such as: 
// Shows the FastLed's light strip

**Review by yelisetti — APPROVED** [review]
Changes look good!

**Comment by yelisetti** [comment]
Merged!

**Pull request merged**